step.go

```diff
@@ -0,0 +1,487 @@
+			Configuration:             input.Configuration,
+			CodeSigningAuthSource:     input.CodeSigningAuthSource,
+			RegisterTestDevices:       input.RegisterTestDevices,
+			MinDaysProfileValid:       input.MinDaysProfileValid,
+			TeamID:                    input.TeamID,
+			CertificateURLList:        input.CertificateURLList,
+			CertificatePassphraseList: input.CertificatePassphraseList,
+			KeychainPath:              input.KeychainPath,
+			KeychainPassword:          input.KeychainPassword,
+			BuildURL:                  input.BuildURL,
+			BuildAPIToken:             input.BuildAPIToken,
+			VerboseLog:                input.VerboseLog,
+		}, xcodebuildVersion.MajorVersion, logger, factory)
+		if err != nil {
+			return Config{}, err
+		}
+		codesignManager = &codesignMgr
+	}
+
+	return Config{
+		ProjectPath:            absProjectPath,
+		Scheme:                 input.Scheme,
+		Configuration:          input.Configuration,
+		Destination:            input.Destination,
+		XCConfigContent:        input.XCConfigContent,
+		XcodebuildOptions:      customOptions,
+		XCPretty:               input.LogFormatter == "xcpretty",
+		CodesignManager:        codesignManager,
+		OutputDir:              absOutputDir,
+		XcodebuildMajorVersion: int(xcodebuildVersion.MajorVersion),
+		CacheLevel:             input.CacheLevel,
+		SwiftPackagesPath:      swiftPackagesPath,
+	}, nil
+}
+
+func (b TestBuilder) InstallDependencies(useXCPretty bool) error {
+	if !useXCPretty {
+		return nil
+	}
+
+	fmt.Println()
+	log.Infof("Checking if output tool (xcpretty) is installed")
+	formatter := xcpretty.NewXcpretty()
+
+	installed, err := formatter.IsInstalled()
+	if err != nil {
+		return err
+	}
+
+	if !installed {
+		log.Warnf("xcpretty is not installed")
+		fmt.Println()
+		log.Printf("Installing xcpretty")
+
+		cmdModelSlice, err := formatter.Install()
+		if err != nil {
+			return fmt.Errorf("failed to create xcpretty install commands: %w", err)
+		}
+
+		for _, cmd := range cmdModelSlice {
+			if err := cmd.Run(); err != nil {
+				return fmt.Errorf("failed to run xcpretty install command (%s): %w", cmd.PrintableCommandArgs(), err)
+			}
+		}
+	}
+
+	xcprettyVersion, err := formatter.Version()
+	if err != nil {
+		return fmt.Errorf("failed to get xcpretty version: %w", err)
+	}
+
+	log.Printf("- xcpretty version: %s", xcprettyVersion.String())
+	return nil
+}
+
+type RunOut struct {
+	XcodebuildLog string
+	BuiltTestDir  string
+	XctestrunPth  string
+	SYMRoot       string
+}
+
+func (b TestBuilder) Run(cfg Config) (RunOut, error) {
+	// Automatic code signing
+	authOptions, err := b.automaticCodeSigning(cfg.CodesignManager)
+	if err != nil {
+		return RunOut{}, err
+	}
+	defer func() {
+		if authOptions != nil && authOptions.KeyPath != "" {
+			if err := os.Remove(authOptions.KeyPath); err != nil {
+				log.Warnf("failed to remove private key file: %s", err)
+			}
+		}
+	}()
+
+	// Build for testing
+	log.Infof("Build:")
```

Comment:
NOTE: add a fmt.Println() before, for better seperated log output. (Same for Export phase.)